components/url_sanitizer/browser/url_sanitizer_service.cc

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/url_sanitizer/browser/url_sanitizer_service.h"
+
+#include <memory>
+#include <vector>
+
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "base/task/task_runner_util.h"
+#include "base/task/thread_pool.h"
+#include "base/values.h"
+#include "extensions/common/url_pattern.h"
+#include "url/gurl.h"
+
+namespace brave {
+
+namespace {
+bool CreateURLPatternSetFromValue(const base::Value* value,
+                                  extensions::URLPatternSet* result) {
+  if (!value || !value->is_list())
+    return false;
+  std::string error;
+  bool valid = result->Populate(
+      base::Value::AsListValue(*value),
+      URLPattern::SCHEME_HTTP | URLPattern::SCHEME_HTTPS, false, &error);
+  if (!valid) {
+    VLOG(1) << "Unable to create url pattern:" << error;
+  }
+  return valid;
+}
+
+bool CreateParamsList(const base::Value* value,
```

Comment:
I think we could have `absl::optional<base::flat_set<std::string>>` as the return type, rather than using an out param, which is discouraged.

Reply:
done